packages/http/libraries/core/src/http/adapter/InversifyHttpAdapter.ts

```diff
@@ -205,67 +205,79 @@ export abstract class InversifyHttpAdapter<
+      if (controllerMethodParameterMetadata === undefined) {
+        handlerParams.push(undefined);
+      } else if (
+        controllerMethodParameterMetadata.parameterType ===
+        RequestMethodParameterType.BODY
+      ) {
+        handlerParams.push(
+          await this._getBody(
+            request,
+            controllerMethodParameterMetadata.parameterName,
+          ),
+        );
+      } else {
+        switch (controllerMethodParameterMetadata.parameterType) {
+          case RequestMethodParameterType.PARAMS:
+            handlerParams.push(
+              this._getParams(
                 request,
                 controllerMethodParameterMetadata.parameterName,
-              );
-            }
-            case RequestMethodParameterType.QUERY: {
-              return this._getQuery(
+              ),
+            );
+            break;
+          case RequestMethodParameterType.QUERY:
+            handlerParams.push(
+              this._getQuery(
                 request,
                 controllerMethodParameterMetadata.parameterName,
-              );
-            }
-            case RequestMethodParameterType.HEADERS: {
-              return this._getHeaders(
+              ),
+            );
+            break;
+          case RequestMethodParameterType.HEADERS:
+            handlerParams.push(
+              this._getHeaders(
                 request,
                 controllerMethodParameterMetadata.parameterName,
-              );
-            }
-            case RequestMethodParameterType.COOKIES: {
-              return this._getCookies(
+              ),
+            );
+            break;
+          case RequestMethodParameterType.COOKIES:
+            handlerParams.push(
+              this._getCookies(
                 request,
                 response,
                 controllerMethodParameterMetadata.parameterName,
-              );
-            }
-            case RequestMethodParameterType.CUSTOM: {
-              return controllerMethodParameterMetadata.customParameterDecoratorHandler?.(
+              ),
+            );
+            break;
+          case RequestMethodParameterType.NEXT:
+            handlerParams.push(next);
+            break;
+          case RequestMethodParameterType.REQUEST:
+            handlerParams.push(request);
+            break;
+          case RequestMethodParameterType.RESPONSE:
+            handlerParams.push(response);
+            break;
+          case RequestMethodParameterType.CUSTOM: {
+            handlerParams.push(
+              controllerMethodParameterMetadata.customParameterDecoratorHandler?.(
                 request,
                 response,
-              );
-            }
-            case RequestMethodParameterType.NEXT: {
-              return next;
-            }
+              ),
+            );
+            break;
           }
-        },
-      ),
-    );
+        }
+      }
+    }
+
+    return handlerParams;
```

Comment:
This code changes are done so fastify can integration can be fixed, but then it's coupled to fastify, for awaiting the body could be harmful in other integration.

I understand awaiting some promise like objects like the fastify `Response` is not what we want and might have undesired side effects. Specific adapters are the ones who know this, not the abstraction.

Does it make sense to you? If so, I have a couple of ideas of how to implement this in a way we depend on an specific adapter settings